client/app/visualizations/chart/index.js

```diff
@@ -216,7 +217,16 @@ function ChartEditor(ColorPalette, clientConfig) {
         if (old !== undefined) {
           unsetColumn(old);
         }
-        if (value !== undefined) { setColumnRole('x', value); }
+        if (value !== undefined) {
+          setColumnRole('x', value);
+        }
+
+        // If every x value is a datetime then set the x-axis type to datetime.
+        const chartData = scope.queryResult.getData();
+        const isDatetimeAxis = chartData.every(dataPoint => moment(dataPoint[value]).isValid());
```

Comment:
This won't catch YYYYMMDD dates, I think.

Reply:
The above change in `client/app/services/query-result.js` stores the query result with `YYYYMMDD` format as a moment. Here we are just validating that every x value is a moment.

But now that we're here, I realized `dataPoint[value]` is already a moment so it can be checked for `isValid()` directly. I'll update that.

Reply:
Ah, makes sense. Leave it as-is then, since non-datetime value won't have an `isValid` method.